### PR TITLE
kv: migrate test framework to testify

### PR DESCRIPTION
**Pull request by LLThomas**
Signed-off-by: LLThomas <zs033@qq.com>

### What problem does this PR solve?

Issue Number: Ref #4813

### What is changed and how does it work?

Just as the title says.
```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: server/storage/kv/kv_test.go [hunk 1]
```diff
@@ -23,83 +23,79 @@ import (
...
 	kv := NewMemoryKV()
-	s.testReadWrite(c, kv)
-	s.testRange(c, kv)
+	testReadWrite(t, kv)
+	testRange(t, kv)
 }
 
-func (s *testKVSuite) testReadWrite(c *C, kv Base) {
+func testReadWrite(t *testing.T, kv Base) {
+	re := require.New(t)
```
Comment (JmPotato) [inline server/storage/kv/kv_test.go:71]: What about just passing `re *require.Assertions` here to avoid unnecessary `require.New(t)`?

**Review by JmPotato — COMMENTED** [review]
The rest LGTM.

File: server/storage/kv/kv_test.go (hunk 1 above)
Comment (LLThomas) [reply server/storage/kv/kv_test.go:71]: Good idea~

**Review by JmPotato — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**